src/restore.ts

```diff
@@ -4,69 +4,129 @@ import * as exec from "@actions/exec";
 import * as process from "process";
 import * as cache from "@actions/cache";
 
-// based on https://cristianadam.eu/20200113/speeding-up-c-plus-plus-github-actions-using-ccache/
+const SELF_CI = process.env["CCACHE_ACTION_CI"] === "true"
 
-async function install() {
-  if (process.platform === "darwin") {
-    await exec.exec("brew install ccache");
-  } else {
-    await exec.exec("sudo apt-get install -y ccache");
-  }
-}
+// based on https://cristianadam.eu/20200113/speeding-up-c-plus-plus-github-actions-using-ccache/
 
-async function restore() {
-  let restoreKey = `ccache-`;
+async function restore(ccacheVariant : string) : Promise<void> {
+  const inputs = {
+    primaryKey: core.getInput("key"),
+    // https://github.com/actions/cache/blob/73cb7e04054996a98d39095c0b7821a73fb5b3ea/src/utils/actionUtils.ts#L56
+    restoreKeys: core.getInput("restore-keys").split("\n").map(s => s.trim()).filter(x => x !== "")
+  };
 
-  let inputKey = core.getInput("key");
-  if (inputKey) {
-    restoreKey += `${inputKey}-`;
-  }
+  const keyPrefix = ccacheVariant + "-";
+  const primaryKey = inputs.primaryKey ? keyPrefix + inputs.primaryKey + "-" : keyPrefix;
+  const restoreKeys = inputs.restoreKeys.map(k => keyPrefix + k + "-")
+  const paths = [`.${ccacheVariant}`];
```

Comment:
this does not look backwards compatible to me, but produces new keys?

Reply:
True.